src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -85,6 +87,56 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+#ifdef CEPH_HAVE_SPLICE
+static bool tcp_pipe(int& source_fd, int& sink_fd) {
+  source_fd = -1;
+  sink_fd = -1;
+  int server_fd = -1;
+  struct sockaddr_in server_addr;
+  memset(&server_addr, 0, sizeof(struct sockaddr_in));
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_addr.s_addr = INADDR_ANY;
+  server_addr.sin_port = 0;
+  try {
+    server_fd = socket(AF_INET, SOCK_STREAM, 0);
+    if (server_fd < 0) throw (__LINE__);
```

Comment:
Nit: even one-line ```if``` statements should be broken up over multiple lines and surrounded by braces. I think that will help with the readability issues here.